### ISSUE TITLE
BulkInsert fails when FK is a shadow property in EFCore 2.1 Preview

### ISSUE THREAD

**Issue by mmiller678** [zzzprojects/EntityFramework-Extensions#94]
System.Data.SqlClient.SqlException : The MERGE statement conflicted with the FOREIGN KEY constraint "FK_DataValue_DataFieldID_DataField_DataFieldID". The conflict occurred in database "UnitTest", table "dbo.DataField", column 'DataFieldID'.

**Domain Model**
```
    public class DataField
    {
        public virtual long Id {get; set; }
        public virtual string Name { get; set; }
    }
    public class DataValue
    {
        public virtual long Id {get; set; }
        public virtual DataField Field { get;set; }
        public virtual string Value { get; set; }
    }
```
**Configurations (relevant parts)**
```
DataField
            builder.HasKey(x => x.Id);

            builder.Property(x => x.Id)
                .HasColumnName("DataFieldID")
                .UseSqlServerIdentityColumn()
                .ValueGeneratedOnAdd();

           builder.Property(x => x.Name)
                .HasColumnName("Name");

DataValue
            builder.HasKey(x => x.Id);

            builder.Property(x => x.Id)
                .HasColumnName("DataValueID")
                .UseSqlServerIdentityColumn()
                .ValueGeneratedOnAdd();

            builder.Property(x => x.Value)
                .HasColumnName("Value");

            builder.HasOne(x => x.Field)
                .WithMany()
                .IsRequired()
                .HasForeignKey("DataFieldId"); //Shadow Property
```
**BulkInsert of DataValue entities**
```
Context.BulkInsert(values, x =>
            {
                x.BatchSize = batchSize; //1000
                x.IncludeGraph = false;
             });
```
With the SQL Server Profiler I see a temp table is created for bulk insert then a merged into DataValue table.  Is the bulk insert  properly setting the DataFieldID value using the mapping provided.  Unlike in EF6 there is no way to define the FK column name on a One to Many relationship mapping.

**Comment by JonathanMagnan** [comment]
Hello @mmiller678 ,

We released around 10 minutes ago a new version (v2.1.5) fixing an issue with `Shadow Property`.

Do you think you could try it and let us know if that's working? Otherwise, we will look at it to fix it.

Best Regards,

Jonathan

**Comment by mmiller678** [comment]
I am getting different exception now:

System.MissingMethodException : Method not found: 'Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IValueGenerationManager Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.get_ValueGeneration()'.
   at .`1.( )
   at .Read()
   at .()
   at System.Data.SqlClient.SqlBulkCopy.ReadFromRowSourceAsync(CancellationToken cts)
   at System.Data.SqlClient.SqlBulkCopy.WriteToServerInternalAsync(CancellationToken ctoken)
   at System.Data.SqlClient.SqlBulkCopy.WriteRowSourceToServerAsync(Int32 columnCount, CancellationToken ctoken)
   at System.Data.SqlClient.SqlBulkCopy.WriteToServer(IDataReader reader)
   at .( , SqlConnection , SqlTransaction , Int32 )
   at .( )
   at .(List`1 )
   at Z.BulkOperations.BulkOperation.()
   at .[](DbContext this, IEnumerable`1 , Action`1 )
   at **My Code StackTrace Removed**

**Comment by JonathanMagnan** [comment]
Hello @mmiller678 ,

The v2.1.6 has been released.

The issue was caused because of a method in the version 2.x was no longer existing in the preview version. 

We tested the scenario you provided and everything worked as expected.

Let me know if that's fixed on your side.

Best Regards,

Jonathan

**Comment by mmiller678** [comment]
I tested quickly and it seems to be working - if I have other issues I will open up a new issue.

**Comment by sandrohanea** [comment]
I have the same problem with SQLite and EF Core.
I tried to create a shadow primary key using the next mappings:

            builder.Property<Guid>(keyName).ValueGeneratedOnAdd();
            builder.HasKey(keyName);

When trying to bulk insert some entities with some related navigation properties having that shadow PK, I got an SQLite FK constraint error.

Code used to bulk insert:

 this.dbContext.BulkInsert(entitiesToAdd, options =>
            {
                options.IncludeGraph = true;
            });
I tried to disable primary keys before inserting but the error is still here.
"PRAGMA foreign_keys = 0;";

Stacktrace of error:
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 19: 'FOREIGN KEY constraint failed'.
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at .(DbCommand , BulkOperation )
   at .( , DbCommand )
   at .Execute(List`1 actions)
   at .(List`1 )
   at Z.BulkOperations.BulkOperation.Execute()
   at Z.BulkOperations.BulkOperation.BulkInsert()
   at .[](DbContext this, IEnumerable`1 , Action`1 )
   at .(DbContext this, StateManager , IReadOnlyList`1 , Action`1 )
   at .(DbContext this, StateManager , IReadOnlyList`1 , Action`1 )
   at .(DbContext this, Action`1 )
   at DbContextExtensions.BulkSaveChanges(DbContext this, Action`1 options)
   at .[](DbContext this, IEnumerable`1 , Action`1 )
   at DbContextExtensions.BulkInsert[T](DbContext this, IEnumerable`1 entities, Action`1 options)
